src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java

```diff
@@ -60,9 +58,30 @@ private BeanDefinition createBeanDefinition(Class<?> aClass) {
                 .withScope(getBeanScope(aClass))
                 .withIsLazy(isLazy(aClass))
                 .withIsPrimary(isPrimary(aClass))
+                .withDependsOn(getDependsOn(aClass))
                 .createBeanDefinitionImpl();
     }
 
+    private Class<?>[] getDependsOn(Class<?> aClass) {
+        var constructors = aClass.getDeclaredConstructors();
+        var constrCount = constructors.length;
+        if (constrCount > 1) {
+            throw new ClassConctructorException("Class have more than one constructor");
```

Comment:
Id suggest to add some specific information, like class name or so

Reply:
fixed